### PR TITLE
rename writeStringToBuffer and some more documentation rework.

### PR DESCRIPTION
**Pull request by ProAce**
fix for https://github.com/yaacov/ArduinoModbusSlave/issues/52

### CODE REVIEW

File: src/ModbusSlave.cpp [hunk 1]
```diff
@@ -462,31 +462,36 @@ uint8_t Modbus::writeRegisterToBuffer(int offset, uint16_t value)
...
-    if ((index + length) > _responseBufferLength - MODBUS_CRC_LENGTH)
+    // Check if the array fits in the remaining space of the response.
+    if ((index + (length * 2)) > _responseBufferLength - MODBUS_CRC_LENGTH)
     {
+        // If not return an exception.
         return STATUS_ILLEGAL_DATA_ADDRESS;
     }
 
-    memcpy(_responseBuffer + index, str, length);
+    for (int i = 0; i < length; i++)
```
Comment (yaacov) [inline src/ModbusSlave.cpp]: hmm, a loop instead of memcpy ? are you sure ...
can you explain in a remark what is happening and continue useing memcpy ?

Comment (ProAce) [reply src/ModbusSlave.cpp]: This is done to enable passing a uint16_t array. As far as I know there is no suitable way to split a uint16_t into two uint8_t's and copy it to an array via memcpy.

Comment (yaacov) [reply src/ModbusSlave.cpp]: where does the endiness comes from ?
if you consider using `memcpy` vs `loop`, the difference is the endiness,
is it a problem you can solve on generating side ?
 
``` c
#include<stdio.h>
#include<stdint.h>
#include<string.h>

void main() {
    int length = 5;
    uint8_t _responseBuffer[] = {0,0,0,0,0,0,0,0,0,0};
    uint16_t str[] = {(5 << 8) + 1, (4 << 8) + 2,(3 << 8) + 3, (2 << 8) + 4, (1 << 8) + 5};

    printf ("copy using loop\n");
    for (int i = 0; i < length; i++)
    {
        _responseBuffer[(i * 2)] = str[i] >> 8;
        _responseBuffer[(i * 2) + 1] = str[i] & 0xFF;
    }
    for (int i = 0; i < length; i++)
    {
        printf ("%d, %d\n",
            _responseBuffer[(i * 2)],
            _responseBuffer[(i * 2) + 1]);
    }

    printf ("copy using memcpy\n");
    memcpy((uint8_t*)_responseBuffer, str, length * 2);
    for (int i = 0; i < length; i++)
    {
        printf ("%d, %d\n",
            _responseBuffer[(i * 2)],
            _responseBuffer[(i * 2) + 1]);
    }

    return;
}
```

Comment (ProAce) [reply src/ModbusSlave.cpp:490]: I didn't know that was possible with memcpy, but when running your example it seems to change the endianess of the values. Modbus expects a Big endian for the 16 bits values, and when using memcpy it switches it to little endian.

I'm also not fond of using a loop, but it seems (to me) to be the best readable and predictable option.

Comment (yaacov) [reply src/ModbusSlave.cpp:490]: ok :-)

actually, I just remember why this method exist :-)
it was made to store r/w **stings**, like maintenance dates, versions, address of device, etc...
the idea was that users could store strings in the registers space :rainbow:

**Comment by yaacov** [comment]
Thanks :unicorn:

File: src/ModbusSlave.cpp (hunk 1 above)
Comment (ProAce) [reply src/ModbusSlave.cpp:490]: That would explain the name and could be a useful application. Having both would then be the ideal way.

Comment (yaacov) [reply src/ModbusSlave.cpp:490]: The importance of good method comments ... :palm_tree: :fox_face:

**Pull request merged**